services/QuillLMS/app/services/classlink_integration/enrollment_persister.rb

```diff
@@ -49,7 +51,10 @@ def run
       ErrorNotifier.report(e, { message: msg })
     end
 
-    def create_classroom_teacher(user, classroom)
+    def create_teacher_linkages(user, classroom)
```

Comment:
Since we're potentially creating many objects here, which might raise an exception, I would wrap this in a `ActiveRecord::Base.transaction`